src/ert/job_runner/job.py

```diff
@@ -108,10 +114,13 @@ def run(self):
                 #
                 # See https://github.com/giampaolo/psutil/issues/1044#issuecomment-298745532  # noqa
                 memory = 0
-            if memory > max_memory_usage:
-                max_memory_usage = memory
+            max_memory_usage = max(memory, max_memory_usage)
 
-            yield Running(self, max_memory_usage, memory)
+            if abs(memory - last_memory_usage) >= self.MEMORY_INCREASE_THRESHOLD:
+                last_memory_usage = memory
+                yield Running(self, max_memory_usage, memory)
+            else:
+                yield None
```

Comment:
Why does it need to yield None? Can it not just continue?

Reply:
There are at least two reasons:
1. We sort of keep it alive (while yielding control to the primary `for loop` in `cli.py`)
2. Makes it much easier to test since we get to know that there was no change as it yields None. Otherwise, it becomes a blocker.